conans/cli/commands/search.py

```
@@ -19,9 +19,16 @@ def search(conan_api, parser, *args, **kwargs):
                         help="Remote to search. Accepts wildcards. To search in all remotes use *")
     args = parser.parse_args(*args)
 
-    remotes = conan_api.get_active_remotes(args.remote)
+    remotes, error = conan_api.get_active_remotes(args.remote)
+    if error:
+        raise ConanException(error)
```

If the API method returns an error, why delegate the exception to the caller? Let the API method raise the exception instead of returning a tuple